### DISCUSSION TITLE
not able to customize the position of the Drawer component

### DISCUSSION THREAD

**Discussion by programerBlueGrid** [chakra-ui/chakra-ui#4356] [Q&A]
Hello, 
I hope someone can help me, I am trying to customize the DrawerContent component that is one the parts of Drawer component that has placement on the right, so it appears from the right side and I am trying to position the drawer to be 20% from the top of page, not to go all way from the top but from position I specify. I have followed the instructions, adding components in extend theme and adding styles in baseStyle but it doesn't work. However it works if I add inline styles to DrawerContent like <DrawerContent style={{ top: "20%" }}> it works  but I would like to do this using extend theme. Here is the code:
```
 export const DrawerStyles = {
  // style object for base or default style
  baseStyle: {
    top: '20%'	
  },
  // styles for different sizes ("sm", "md", "lg")
  sizes: {},
  // styles for different visual variants ("outline", "solid")
  variants: {
  },
  // default values for `size` and `variant`
  defaultProps: {
    size: "",
    variant: "",
  },
};

```

```
import { extendTheme } from "@chakra-ui/react";
import { DrawerStyles as DrawerContent } from "./custom-components/DrawerStyles";

export const theme = extendTheme({
  styles: {
    global: {
      body: {
        backgroundColor: "white.100",
      },
    },
  },
  components: {
    DrawerContent,
  },

  fonts: {
    body: "Lato, sans-serif",
    heading: "Lato, sans-serif",
    mono: "Lato, monospace",
  },
  colors: {
    green: {
      100: "#80c740",
      600: "#0A864F",
    },

    red: {
      100: "#ff0000",
    },

    grey: {
      100: "#d3d3d3",
      200: "#a6a6a6",
      600: "#888888",
    },

    white: {
      50: "#ffffff",
      100: "#fafdf9",
      200: "#eff9ec",
      300: "#dff2d9",
    },
  },
});

```

Then I am just importing it in my file import {  DrawerContent,  Drawer, DrawerBody, DrawerFooter, DrawerHeader } from "@chakra-ui/react"

**Comment by primos63** [comment]
You need to refer to the [theme source](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/drawer.ts) for the component to figure out how to structure your theme extension.
Drawer is a multi-part component the baseStyle is
```javascript
const baseStyle = (props: Record<string, any>) => ({
  overlay: baseStyleOverlay,
  dialogContainer: baseStyleDialogContainer,
  dialog: baseStyleDialog(props),
  header: baseStyleHeader,
  closeButton: baseStyleCloseButton,
  body: baseStyleBody,
  footer: baseStyleFooter,
})
```
What you need in your theme is
```javascript
  components: {
    Drawer: {
      baseStyle: {
        dialogContainer: {
          height: '80vh',
          top: '20%',
        },
      },
    },
  },
```
This will only affect the container. The `DrawerContent` is represented by a div element with a class of "chakra-slide" which has a style attribute set to `position: fixed; right: 0; top: 0; bottom: 0; width: 100%` along with a transform. The only way I could find to override it is by adding style to the DrawContent `<DrawerContent style={{ top: '20%' }}>`.

There could be a better way, but I just don't see it right now.
